src/app/pages/storage/volumes/volumes-list/volumes-list.component.ts

```diff
@@ -294,14 +294,25 @@ export class VolumesListTableConfig implements InputTableConf {
         });
       }
 
-    } else if (rowData.encrypt === 1 && rowData.is_decrypted) {
-      actions.push({
-        label: T("Encryption Key"),
-        onClick: (row1) => {
-          this._router.navigate(new Array('/').concat(
-            ["storage", "pools", "createkey", row1.id]));
-        }
-      });
+    } else if (rowData.encrypt === 1 && rowData.is_decrypted ) {
+      if (self.parentVolumesListComponent.systemdatasetPool != rowData.name) {
+        actions.push({
+          label: T("Encryption Key"),
+          onClick: (row1) => {
+            this._router.navigate(new Array('/').concat(
+              ["storage", "pools", "createkey", row1.id]));
+          }
+        });
+      } else {
```

Comment:
Would this still allow downloading key from geli pools which do not have system dataset ?

Reply:
yes, they can go to the encryption management page and the option is available there although if you like I can make this option show up for any encrypted pool